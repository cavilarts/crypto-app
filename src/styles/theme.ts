import { createTheme } from "@mui/material/styles";

// Create a theme instance.
const theme = createTheme({
  palette: {
    primary: {
      main: "#232323",
      dark: "#002884",
    },
    secondary: {
      main: "#b453f5",
    },
    error: {
      main: "#e63a2b",
    },
  },
});

export default theme;
