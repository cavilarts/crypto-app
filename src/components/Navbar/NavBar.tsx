import { FC, useCallback, useState } from "react";
import { AppBar, Hidden, IconButton, Toolbar, Typography } from "@mui/material";
import { Menu } from "@mui/icons-material";
import Link from "next/link";
import { DrawerNav, MenuLink } from "./NavBar.styles";

export const Navbar: FC = () => {
  const [open, setOpen] = useState(false);

  const menuOnClick = useCallback(() => setOpen(!open), []);

  return (
    <>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" component="h5" sx={{ flexGrow: 1 }}>
            Crypto News
          </Typography>
          <Hidden smUp>
            <IconButton
              size="large"
              edge="start"
              color="inherit"
              aria-label="Menu"
              onClick={menuOnClick}
              sx={{ mr: 2 }}
            >
              <Menu />
            </IconButton>
          </Hidden>
          <Hidden smDown>
            <Link href="/">
              <MenuLink>Crypto Currencies</MenuLink>
            </Link>
            <Link href="/">
              <MenuLink>Exchanges</MenuLink>
            </Link>
            <Link href="/">
              <MenuLink>News</MenuLink>
            </Link>
          </Hidden>
        </Toolbar>
      </AppBar>
      <DrawerNav anchor="right" open={open} onClose={() => setOpen(false)}>
        <Link href="/">
          <MenuLink>Crypto Currencies</MenuLink>
        </Link>
        <Link href="/">
          <MenuLink>Exchanges</MenuLink>
        </Link>
        <Link href="/">
          <MenuLink>News</MenuLink>
        </Link>
      </DrawerNav>
    </>
  );
};
