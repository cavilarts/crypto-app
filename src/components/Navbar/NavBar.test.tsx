/**
 * @jest-environment jsdom
 */
import { fireEvent, render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import {
  createTheme,
  ThemeProvider as MuiThemeProvider,
} from "@mui/material/styles";

import { Navbar } from "./NavBar";
import theme from "../../styles/theme";

import mediaQuery from "css-mediaquery";

function createMatchMedia(width) {
  return (query) => ({
    matches: mediaQuery.match(query, {
      width,
    }),
    addListener: () => {},
    removeListener: () => {},
  });
}

const customTheme = createTheme({
  ...theme,
  components: {
    MuiUseMediaQuery: {
      defaultProps: {
        noSsr: true,
      },
    },
  },
});

const setup = () =>
  render(
    <MuiThemeProvider theme={customTheme}>
      <ThemeProvider theme={customTheme}>
        <Navbar />
      </ThemeProvider>
    </MuiThemeProvider>
  );

describe("<Navbar />", () => {
  beforeAll(() => {
    window.matchMedia = createMatchMedia(380);
  });

  it("should match the snapshot", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it("should open the drawer", () => {
    const { getByRole, container } = setup();
    const menuButton = getByRole("button", { name: "Menu" });

    expect(menuButton).toBeInTheDocument();

    fireEvent.click(menuButton);

    expect(container).toMatchSnapshot();

    fireEvent.click(document.body);
  });
});
