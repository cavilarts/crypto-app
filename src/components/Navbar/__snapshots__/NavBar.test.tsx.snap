// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1go5nc5-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        Crypto News
      </h5>
      <button
        aria-label="Menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`<Navbar /> should open the drawer 1`] = `
<div
  aria-hidden="true"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1go5nc5-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        Crypto News
      </h5>
      <button
        aria-label="Menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
